tests/db_utils/db_setup.py

```diff
@@ -69,49 +71,59 @@ async def setup_db(db_url: str, db_name: str, data_dir: str):
-        df.rename(columns={column: column_info.name}, inplace=True)
-
-        if column_info.refers_to is not None:
-          fk_ref_table_name = column_info.refers_to[0]
-          if fk_ref_table_name not in fk_constraints:
-            fk_constraints[fk_ref_table_name] = {'cols': [], 'cols_refs': []}
-          # both tables will have same column name
-          fk_constraints[fk_ref_table_name]['cols'].append(column_info.name)
-          fk_constraints[fk_ref_table_name]['cols_refs'].append(
-            f"{column_info.refers_to[0]}.{column_info.refers_to[1]}")
-
-      multi_table_fk_constraints = []
-      for tbl, fk_cons in fk_constraints.items():
-        multi_table_fk_constraints.append(ForeignKeyConstraint(fk_cons['cols'], fk_cons['cols_refs']))
-
-      _ = sqlalchemy.Table(table_name, metadata, *[
-        sqlalchemy.Column(c_info.name, c_info.dtype, primary_key=c_info.is_primary_key) for c_info in columns_info
-      ], *multi_table_fk_constraints)
-
-    await conn.run_sync(lambda conn: metadata.create_all(conn))
+  try:
+    async with engine.begin() as conn:
+      await drop_all_tables(conn)
+      metadata = MetaData(bind=conn)
+      # Create tables
+      for csv_fname in gt_csv_fnames:
+        base_fname = os.path.basename(csv_fname)
+        table_name = base_fname.split('.')[0]
+        df = pd.read_csv(csv_fname)
+        columns_info = []
+        fk_constraints = {}
+        for column in df.columns:
+          column_info = extract_column_info(table_name, column)
+          dtype = python_type_to_sqlalchemy_type(df[column].dtype)
+          if dtype == sqlalchemy.String:
+            # TODO: VAR CHAR lenth should be based on the number of characters
+            if column_info.is_primary_key or column_info.refers_to:
+              column_info.dtype = sqlalchemy.VARCHAR(256)
+            else:
+              column_info.dtype = sqlalchemy.TEXT
+          else:
+            column_info.dtype = dtype
+          columns_info.append(column_info)
+          df.rename(columns={column: column_info.name}, inplace=True)
+
+          if column_info.refers_to is not None:
+            fk_ref_table_name = column_info.refers_to[0]
+            if fk_ref_table_name not in fk_constraints:
+              fk_constraints[fk_ref_table_name] = {'cols': [], 'cols_refs': []}
+            # both tables will have same column name
+            fk_constraints[fk_ref_table_name]['cols'].append(column_info.name)
+            fk_constraints[fk_ref_table_name]['cols_refs'].append(
+              f"{column_info.refers_to[0]}.{column_info.refers_to[1]}")
+
+        multi_table_fk_constraints = []
+        for tbl, fk_cons in fk_constraints.items():
+          multi_table_fk_constraints.append(ForeignKeyConstraint(fk_cons['cols'], fk_cons['cols_refs']))
+
+        _ = sqlalchemy.Table(
+            table_name,
+            metadata,
+            *[sqlalchemy.Column(c_info.name, c_info.dtype, primary_key=c_info.is_primary_key, autoincrement=False)
```

Comment:
Configure the 'autoincrement' attribute to 'False' to prevent the unintended conversion of '0' to '1' in MySQL database, which can lead to an integrity error.